### ISSUE TITLE
Ability to use url in $include and $file directives in config.yaml

### ISSUE THREAD

**Issue by prkeshri** [backstage/backstage#8176]
Hello

I would like to suggest an addition where the $include and $file directives can be supplied with a url to load the contents (Currently, I see that only local files are supported).

## Feature Suggestion

This will be helpful in situations where the config or contents are not present in the repository but demands loading from an external url.

## Possible Implementation

I will be happy to contribute. I would require an access/link to the code for @backstage/config-loader (Unfortunately, I could not find it). Once this is accepted and the link is provided, I will add my implementation and raise PR.
So, an $include or $file can have url:<url> . Because of the url prefix, rest of the value will be a url from where the data has to be fetched.
## Context

**What are you trying to accomplish?** Ability to load config/ file from url apart from the disk.

**Comment by adamdmharvey** [comment]
Hi @prkeshri ! 👋🏻 

To find the packages, the convention is,

* `@backstage/plugin-myname` pages all exist in the subdirectory `plugins/`
* Core packages, those starting with `@backstage/` and not including the word plugin, all exist in `packages/`

The package you're looking for can be found here:
https://github.com/backstage/backstage/tree/master/packages/config-loader

**Comment by prkeshri** [comment]
> 

Thanks @adamdmharvey . I will have a look and proceed.

**Comment by Rugvip** [comment]
Thank you for the suggesting @prkeshri!

Before proceeding, could you check out the changes in #7340 and see if they fit your need? I'd like to always prefer and existing solution over a new one 😁

**Comment by prkeshri** [comment]
> #7340

Hi @Rugvip 

Thanks for your response. Actually, I was wishing to include remote files/ objects (using $include/ $file) . I have raised PR. Kindly have a look and approve if seems proper.
https://github.com/backstage/backstage/pull/8211

**Comment by prkeshri** [comment]
Hi @Rugvip 

In a different note, regarding the changes in #7340, I can see that there is no way to pass the 'remote' value in loadBackendConfig , also the function loadBackendConfig is not passing this on its own, hence I feel this is a bug. Please guide me regarding the same. In case this is an issue as I feel, I am happy to contribute and fix it.

**Comment by Rugvip** [comment]
@prkeshri see the updated config docs, https://backstage.io/docs/conf/writing#configuration-files. Following #7340 you can now pass URLs as configuration targets, for example the example from the docs:

```bash
yarn start --config ../../app-config.yaml --config https://some.domain.io/app-config.yaml
```

**Comment by prkeshri** [comment]
@Rugvip I am getting the following error when I run the command as above:
```
Backend failed to start up Error: Remote config detected but this feature is turned off
    at loadConfig (webpack-internal:///../config-loader/src/loader.ts:156:11)
    at loadBackendConfig (webpack-internal:///../backend-common/src/config.ts:214:100)
    at async main (webpack-internal:///./src/index.ts:100:18)
```
Please have a look. I tried to inspect and that was my conclusion as mentioned in the above comment. 
Thanks

**Comment by Rugvip** [comment]
@prkeshri Oh yeah sorry, you're right! I think we could perhaps make the reload interval required for now rather than pick a default.

**Comment by prkeshri** [comment]
@Rugvip No worries. I will be raising a PR for the same. Will be happy to contribute :)

**Comment by prkeshri** [comment]
@Rugvip I am done. You may review the same: https://github.com/backstage/backstage/pull/8236/files.

**Comment by Rugvip** [comment]
Fixed by #8406
